src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -167,6 +192,16 @@ private[impl] class InstanceTrackerActor(
     }
   }
 
+  private def updateRepository(repositoryFunc: () => Future[Done], ack: Ack)(implicit ec: ExecutionContext): Future[RepositoryStateUpdated] = {
```

Comment:
This is a very strange recovery code: we write and fail, but then we try to read and if we find the value then everything is ok? Is there an actual ZK scenario that is covered by this? /cc @timcharper 
Otherwise, I'll simply remove all that and say `repo.delete().pipeTo(self)` where we simply abdicate on write requests.

Reply:
based on the comment this is not trying to treat a ZK error but serialization error. Apparently this is trying to treat potential serialization errors that e.g. happened just for some combination of fields.

I would argue that I cleaned the code to the point where it's not such a big problem to leave it here but I won't be against removing that. But if we do that, it is up to discussion whether we don't have to distinguish between ZK errors and serialization errors here and still treat serialization errors as we do now (without abdicating)